PyInstaller/building/datastruct.py

```diff
@@ -50,7 +50,7 @@ def _normentry(self, entry):
             logger.info("TOC found a %s, not a tuple", entry)
             raise TypeError("Expected tuple, not %s." % type(entry).__name__)
         name, path, typecode = entry
-        if typecode in ["BINARY", "DATA"]:
+        if typecode in ["BINARY", "DATA"] and not name.endswith('.typelib'):
             # Normalize the case for binary files and data files only (to avoid duplicates
```

Comment:
What about `... and not os.path.dirname(path) == 'typelibs'`?

Hmm, maybe you should even use a different dirname, maybe `gi_typlibs`? This would state more clearly, where these files belong and reduce the risk of name-clashes. As we are setting `GI_TYPELIB_PATH` in the runtime-hook, we can use any good name.